test/malli/core_test.cljc

```diff
@@ -944,14 +944,14 @@
                                :type :malli.core/invalid-dispatch-value}]}
                     (m/explain schema invalid3)))
 
-      (is (= valid1 (m/parse schema valid1)))
-      (is (= valid2 (m/parse schema valid2)))
+      (is (= (miu/-tagged :sized valid1) (m/parse schema valid1)))
+      (is (= (miu/-tagged :human valid2) (m/parse schema valid2)))
       (is (= ::m/invalid (m/parse schema invalid1)))
       (is (= ::m/invalid (m/parse schema invalid2)))
       (is (= ::m/invalid (m/parse schema invalid3)))
       (is (= ::m/invalid (m/parse schema "not-a-map")))
-      (is (= valid1 (m/unparse schema valid1)))
-      (is (= valid2 (m/unparse schema valid2)))
+      (is (= valid1 (m/unparse schema (m/parse schema valid1))))
+      (is (= valid2 (m/unparse schema (m/parse schema valid2))))
```

Comment:
I would prefer just `(m/unparse schema (miu/-tagged :sized valid1))`, I get a bad vibe from using the SUT when creating the test inputs (even if it is a different part, tested separately).